### PR TITLE
Coordinate

### LINKED ISSUE #4: Create entity and repository for Coordinate

### PR DESCRIPTION
**Pull request by gmonso**

### CODE REVIEW

File: src/main/java/cat/udl/eps/softarch/myroutes/domain/Coordinate.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+@Entity
+@Data
+@EqualsAndHashCode(callSuper = true)
+public class Coordinate extends UriEntity<String> {
```
Comment (rogargon) [inline src/main/java/cat/udl/eps/softarch/myroutes/domain/Coordinate.java]: The id is Long, so it should extend UriEntity<Long>

File: src/main/java/cat/udl/eps/softarch/myroutes/domain/Coordinate.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    @Id
+    @GeneratedValue()
+    private Long id;
+
+    @NotBlank
+    private String coordinate;
+
+    @Override
+    public String getId() {
+        return null;
```
Comment (rogargon) [inline src/main/java/cat/udl/eps/softarch/myroutes/domain/Coordinate.java]: No need to override getId(), remove the code as it is already generated by Lombok and using the appropriate return type, which is Long and not String.

**Review by rogargon — CHANGES_REQUESTED** [review]

File: src/main/java/cat/udl/eps/softarch/myroutes/domain/Coordinate.java (hunk 1 above)
Comment (gmonso) [reply src/main/java/cat/udl/eps/softarch/myroutes/domain/Coordinate.java]: Fixed, now UriEntity is Long

File: src/main/java/cat/udl/eps/softarch/myroutes/domain/Coordinate.java (hunk 2 above)
Comment (gmonso) [reply src/main/java/cat/udl/eps/softarch/myroutes/domain/Coordinate.java]: Fixed, my ide imports a wrong library and generates this function.

**Review by rogargon — APPROVED** [review]

**Pull request merged**